### DISCUSSION TITLE
Pets are not protected (outside claims)

### DISCUSSION THREAD

**Discussion by gnubufferoverflows** [GriefPrevention/GriefPrevention#354] [Q&A] [answered]
### What behaviour is observed:
When you tame a horse (or other pet), any other player can ride that pet without doing the /givepet command. In claimed areas, only the claim owner can ride the pet regardless of who tamed it.
### What behaviour is expected:
When you tame a horse (or any other pet), only the person who tamed the pet can ride it until they do /givepet <player>. The same behavior applies to claimed areas.
### Steps/models to reproduce:
Tame some pets, go on an alt account, and try to ride them.
### Spigot/Craftbukkit/Paper and GriefPrevention version:
[14:36:24 INFO]: This server is running CraftBukkit version git-Spigot-2b0e71c-15da706 (MC: 1.13) (Implementing API version 1.13-R0.1-SNAPSHOT)
[14:36:24 INFO]: Checking version, please wait...
[14:36:25 INFO]: You are 10 version(s) behind
>version GriefPrevention
[14:36:30 INFO]: GriefPrevention version 16.9.alpha

### Stack trace/error
No stack traces or error logs.

### Plugin list (if applicable):
GriefPrevention
PopulationDensity

**Comment by RoboMWM** [comment]
This is expected behavior for PvP-enabled worlds. Big_Scary's reasoning was that people could "park" their pets in other players' claims and they'd have no way to remove them.

Pet protection design is likely going to be somewhat reworked in version 20

**Comment by gnubufferoverflows** [comment]
That's interesting, in my configuration I have the world set to survival and not pvp, which means that those rules shouldn't apply.  I've also tried this on a production server with PvP disabled through server.properties, and survival rules.

**Comment by gnubufferoverflows** [comment]
Config is here: https://gist.github.com/Dogeveloper/512673f0f435b45be823db74ddb225d2

**Comment by RoboMWM** [comment]
Is the behavior as expected on 16.8? I know Big_Scary's pet protection design is a little odd. What I mentioned might've been something that only applies to wolves... and that horses design could be global.

**Comment by gnubufferoverflows** [comment]
In 1.12.2 (GP 16.8) all tameable mobs got pet protection, which overrides land claim permissions in most cases. That way if someone accidentally rides their horse into a land claim, there isn't a problem.

**Comment by RoboMWM** [comment]
> That way if someone accidentally rides their horse into a land claim, there isn't a problem.

Except for admin claims, or so I've heard?

**Comment by gnubufferoverflows** [comment]
I think so, but it would be annoying for admins to extract pets from admin claims if they get stuck. This is especially the case with servers that use PopDensity, because pets teleport along with players if they're teleporting to spawn.

**Comment by RoboMWM** [comment]
Mind running EventDebug? Code hasn't changed for the handled event, so maybe some other event is being fired perhaps? https://github.com/TechFortress/GriefPrevention/blob/bea2fa3d7f0680864338017a5f84a8e5787a3cc4/src/main/java/me/ryanhamshire/GriefPrevention/PlayerEventHandler.java#L1162-L1199

**Comment by RoboMWM** [comment]
Also your reproduction steps seem to be incorrect. You're attempting to ride entities that the alt account doesn't own?

**Comment by gnubufferoverflows** [comment]
Well the correct behavior is that when my alt account rides a pet it doesn't own, that action is cancelled. Anyways, I will try with EventDebug.

**Comment by RoboMWM** [comment]
And 16.8 prevents this behavior?

**Comment by RoboMWM** [comment]
Also, I thought the issue included

> In claimed areas, only the claim owner can ride the pet regardless of who tamed it.

So... is this the case as well, and is not so in 16.8 (with exception of admin claims)?

**Comment by gnubufferoverflows** [comment]
Okay, I tried with EventDebug and these events fired:
```
[18:58:10 INFO]: [EventDebug] Event fired: PlayerInteractAtEntityEvent
getClickedPosition: -0.6982421875,1.1906425952911377,0.2843475043773651
isCancelled: false
getHand: HAND
getRightClicked: CraftHorse{variant=HORSE, owner=CraftPlayer{name=Mrawesomecookie}}
getPlayer: Cateveloper
[18:58:10 INFO]: [EventDebug] Event fired: PlayerInteractEntityEvent
isCancelled: false
getHand: HAND
getRightClicked: CraftHorse{variant=HORSE, owner=CraftPlayer{name=Mrawesomecookie}}
getPlayer: Cateveloper
[18:58:10 INFO]: [EventDebug] Event fired: VehicleEnterEvent
isCancelled: false
getEntered: Cateveloper
getVehicle: CraftHorse{variant=HORSE, owner=CraftPlayer{name=Mrawesomecookie}}
[18:58:10 INFO]: [EventDebug] Event fired: PlayerTeleportEvent
getCause: UNKNOWN
isCancelled: false
getTo: Location{world=CraftWorld{name=world},x=10147.29291839428,y=74.0,z=10237.08324274249,pitch=0.0,yaw=302.69885}
getFrom: Location{world=CraftWorld{name=world},x=10147.29291839428,y=74.0,z=10237.08324274249,pitch=0.0,yaw=235.19922}
getPlayer: Cateveloper
```
When trying to hurt the poor horse:
```

[19:01:18 INFO]: [EventDebug] Event fired: EntityDamageByEntityEvent
getDamager: Cateveloper
isCancelled: false
getDamage: 1.0
getFinalDamage: 1.0
getCause: ENTITY_ATTACK
getEntity: CraftHorse{variant=HORSE, owner=CraftPlayer{name=Mrawesomecookie}}
getEntityType: HORSE
[19:01:18 INFO]: [EventDebug] Event fired: EntityDamageByEntityEvent
getDamager: Cateveloper
isCancelled: false
getDamage: 1.0
getFinalDamage: 1.0
getCause: ENTITY_ATTACK
getEntity: CraftHorse{variant=HORSE, owner=CraftPlayer{name=Mrawesomecookie}}
getEntityType: HORSE
[19:01:18 INFO]: [EventDebug] Event fired: EntityDamageByEntityEvent
getDamager: Cateveloper
isCancelled: false
getDamage: 1.0
getFinalDamage: 1.0
getCause: ENTITY_ATTACK
getEntity: CraftHorse{variant=HORSE, owner=CraftPlayer{name=Mrawesomecookie}}
getEntityType: HORSE
[19:01:18 INFO]: [EventDebug] Event fired: PlayerInteractEvent
isCancelled: true
isBlockInHand: false
getAction: LEFT_CLICK_AIR
getMaterial: AIR
getHand: HAND
getBlockFace: SOUTH
getPlayer: Cateveloper
[19:01:18 INFO]: [EventDebug] Event fired: PlayerAnimationEvent
isCancelled: false
getAnimationType: ARM_SWING
getPlayer: Cateveloper
```

**Comment by gnubufferoverflows** [comment]
It also doesn't work with 16.8.

**Comment by gnubufferoverflows** [comment]
And I think BigScary's original intention was that if the area is
unclaimed, you can't ride or hurt a pet (or otherwise interact with it)
unless you are the owner. If you're in a claim, the pet protection
overrides the claim protection so people don't get their pets stuck. I
don't have any 1.12.2 servers to test how the pet protection works in an
admin claim.

**Comment by gnubufferoverflows** [comment]
In claimed areas, the claim protection applies and not the pet protection.
That behavior is also incorrect.

**Comment by SiuanSong** [comment]
We have been having this issue with dogs, except they can be hit/killed inside and outside of claims.  All nodes in configurations are correct.

**Comment by RoboMWM** [comment]
> We have been having this issue with dogs, except they can be hit/killed inside and outside of claims.

Intended behavior for PvP worlds. Wolves can retaliate, having them protected would make them invincible even if they're attacking.

**Comment by SiuanSong** [comment]
Ah.  Tamed wolves used to be protected not too long ago and we've always had PvP enabled.  Tamed wolves simply weren't able to be hit, bypassing their ability to retaliate against players.

**Comment by RoboMWM** [comment]
That's not true, unless you were using some old version of GP (over 2+ years ago?)

**Comment by NullCase** [comment]
@Dogeveloper here is a 1.12.2 server.

You will almost certainly find everyones pets around spawn. As you say, pets are the owners property and do not respond to the claim owner. So only the pet owner can ride them, stand them up, harm them. And that's true on claims or unclaimed areas.

Wolves, parrots, horses, cats are all protected as pets.

edit: oh, the IP was auto removed. Its Liberty Minecraft's "Old World"

**Comment by RoboMWM** [comment]
@NullCase is the world pvp-enabled? Config for that server and the world the pets are in?

**Comment by RoboMWM** [comment, marked as answer]
According to comments in #295 this is expected behavior.

**Comment by NullCase** [comment]
pvp enabled, yes.

**Comment by TomLewis** [comment]
Pets shouldn't be protected outside of claims, that would give people invincible dogs to guard them, or be able to escape on an invincible horse.
You should look into a pets plugin if you wish to enable this type of gameplay.

**Comment by gnubufferoverflows** [comment]
That's not the point. The point is stop other players from killing them on
non-PvP servers. The pets can still die of natural causes, however.

**Comment by RoboMWM** [comment]
Either create a new issue or update this issue with the correct information, and include both GP config and server version. Also need to test with 16.8 / 16.7.1 on 1.12.2, because afaik horses should be protected outside of claims on a world that does not have PvP enabled (and has PvP rules enabled in config).

**Comment by gnubufferoverflows** [comment]
Sounds good. Also going to try this on 1.13.1 now that it is out.
